app/routes.py

```diff
@@ -1 +1,206 @@
+    task = validate_model(Task,task_id)
+    task.completed_at = datetime.datetime.now()
+    db.session.commit()
+
+    post_to_slack(task)
+
+    return {"task": task.to_dict()}, 200
+
+def post_to_slack(task):
+    path = "https://slack.com/api/chat.postMessage"
+    headers = {"authorization": os.environ.get("SLACK_API_KEY")}
+    data = {
+        "channel": "tasks_notifications",
+        "text" : f"task {task.title} has been completed!"
+    }
+    response = requests.post(path, headers = headers, json = data)
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_title_as_incomplete(task_id):
+    task = validate_model(Task,task_id)
+    task.completed_at = None
+    
+    db.session.commit()
+    return {"task": task.to_dict()}, 200
+
+@goal_bp.route("", methods=["POST"])
+def make_a_goal():
+    try:
+        request_body = request.get_json()
+
+        new_goal = Goal.from_dict(request_body)
+
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return {
+        "goal": new_goal.to_dict()}, 201
+    except KeyError as error:
+        return {"details": "Invalid data"}, 400
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    
+    goal_response = []
+
+    for goal in goals:
+        goal_response.append(goal.to_dict())
+    return jsonify(goal_response)
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+
+    return {"goal" : goal.to_dict()}, 200
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+
+    request_body = request.get_json()
+    
+    try:
+        goal.title = request_body["title"]
+    except:
+        return {"details": "Invalid data"}, 404
+
+    db.session.commit()
+    return  {
+        "goal": goal.to_dict()}, 200
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+    db.session.delete(goal)
+    db.session.commit()
+
+    return  {
+        "details": f"Goal {goal_id} \"{goal.title}\" successfully deleted"
+    }, 200
+
+@goal_bp.route("/<goal_id>/tasks", methods = ["POST"])
+def add_list_of_tasks(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    task_list = request_body.get("task_ids")
+    for task_id in task_list:
+        new_task = validate_model(Task, task_id)
+        new_task.goal_id = goal.goal_id
+        new_task.goal = goal
+    db.session.commit()
+    return {"id": goal.goal_id, "task_ids": task_list}, 200
+
+@goal_bp.route("/<goal_id>/tasks", methods = ["GET"])
+def get_tasks_for_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    task_list = []
+    for task in goal.tasks:
+        task_list.append(task.goal_to_dict())
+    
+    return {
+        "id" : goal.goal_id,
+        "title": goal.title, 
+        "tasks": task_list
+    }
```

Comment:
Well done on this project, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭